cg/services/run_devices/run_names/pacbio.py

```diff
@@ -0,0 +1,20 @@
+from pathlib import Path
+
+from cg.services.run_devices.run_names.service import RunDirectoryNamesService
+
+
+class PacbioRunDirectoryNamesService(RunDirectoryNamesService):
+
+    def get_run_names(self) -> list[str]:
+        """
+        Get all the run names from a run directory, assuming the run directory has only sequencing
+        folders with SMRTcells inside. The run names are in the form <sequencing_run>/<SMRTcell>:
+        Example:
+            r84202_20240913_121403/1_C01
+        """
+        run_names = []
+        for run_folder in Path(self.run_directory).iterdir():
+            if run_folder.is_dir():
```

Comment:
```suggestion
```
Just a suggestion that if we are to be relying on the structure being correct, this check should be unnecessary right? If we want it, it might be that we do not want to trust the structure at all and redesign a bit. But I am not a fan of half measures

Reply:
That is a good point. But, as it is now, if there is a **directory** with an unexpected structure, an incorrect name would be generated and this will raise an error in the post-processing, which will be caught. However, If there is a file in the sequencing directory. the `run_folder.iterdir()` will raise a `NotADirectoryError` which we currently don't catch. This is an easy solution to that.

Reply:
I would say that having a file in this directory is enough cause for concern to not catch the error though? Or why would we want to disregard that?

Reply:
in person: decided to keep it